common/Geometry2d/Point.hpp

```diff
@@ -12,75 +12,65 @@ Simple class to represent a point in 2d space. Uses floating point coordinates
 */
 class Point {
 public:
-    /** the x coordinate */
-    float x;
-
-    /** the y coordinate */
-    float y;
+    const float& x() const { return _x; }
```

Comment:
Maybe add an inline specifier. Its not really necessary, but it would be sort of annoying if the compiler didn't inline for some reason